colossalai/pipeline/schedule/zero_bubble_pp.py

```diff
@@ -39,6 +39,20 @@ def deallocate_output_tensor(out, deallocate_pipeline_outputs=False):
     out.data.untyped_storage().resize_(0)
 
 
+def require_grad(tensor):
+    """Pseudo-deallocate (i.e., set to scalar) the output tensor's '.data' field.
+
+    This method should be called right after the output tensor has been
+    sent to the next pipeline stage. At this point, the output tensor is
+    only useful for its '.grad_fn' field, and not its '.data'.
+    """
+    if tensor is None:
```

Comment:
`if not isinstance(tensor, torch.Tensor):`

Reply:
and this function is redundant, which is defined in `_utils.py`

Reply:
Fixed in https://github.com/hpcaitech/ColossalAI/pull/6065/commits/6ee9584b9a2310bdd556ef32c9901828c2aec04d; Also remove redundant function ;

Reply:
> and this function is redundant, which is defined in `_utils.py`

Fixed in https://github.com/hpcaitech/ColossalAI/pull/6065/commits/6ee9584b9a2310bdd556ef32c9901828c2aec04d; Also remove redundant function ;